### PR TITLE
perf: replace the ipv6 map by an array of struct

### PR DESCRIPTION
**Pull request by Jorropo**
While starting `ipfs` this package used to take ~20ms while initing:
```
init github.com/libp2p/go-libp2p-asn-util @21 ms, 20 ms clock, 2623128 bytes, 892 allocs
```
That is because the map literal is actually creating a new map and adding entries one by one, which for such big map takes a while.
~~This PR moves this to a separate goroutine and use a `sync.Once` to synchronise accesses. Almost completely removing it:~~
```
init github.com/libp2p/go-libp2p-asn-util @8.7 ms, 0.002 ms clock, 128 bytes, 3 allocs
```

Never is this map actually used as a map, the only consumer just iterate over it.
So we can use an array of structs containing `k` and `v` instead, this pr make the init `0.005ms`.

### CODE REVIEW

**Review by willscott — APPROVED** [review]

**Comment by Jorropo** [comment]
@willscott let's not merge this. I have red the code and this change is overkill.

Never is this map actually used as a map, the only consumer just iterate over it.

So we can use an array of structs containing `k` and `v` instead.

**Comment by Jorropo** [comment]
@willscott Update it's now slower `0.005ms` instead of `0.002ms` however it's `0.005ms` of CPU time instead of `0.002ms` of wall time and `20ms` of CPU time.

File: generate/main.go [hunk 1]
```diff
@@ -57,11 +57,11 @@ func writeMappingToFile(f *os.File, m map[string]string, mapName string) {
...
 	printf("package %s\n\n", pkgName)
 	printf("// Code generated by generate/main.go DO NOT EDIT\n")
-	printf("var %s = map[string]string {", mapName)
+	printf("var %s = [...]struct{ cidr, asn string }{", listName)
 	for k, v := range m {
-		printf("\n\t \"%s\": \"%s\",", k, v)
+		printf("\n\t{\"%s\", \"%s\"},", k, v)
 	}
-	printf("\n}")
+	printf("}\n")
```
Comment (willscott) [inline generate/main.go]: do you want a `\n` both in front and after this final closing brace?

Comment (Jorropo) [reply generate/main.go]: Yeah you are right. 
`go fmt` fixed it anyway.

Comment (Jorropo) [reply generate/main.go]: @willscott fixed:
```diff
-	{"2a02:6300::/42", "50128"}}
+	{"2a02:6300::/42", "50128"},
+}
```

**Review by willscott — APPROVED** [review]

**Pull request merged**